src/braket/experimental/autoqasm/operators/logical.py

```diff
@@ -46,3 +136,31 @@ def _oqpy_eq(a: Any, b: Any) -> aq_types.BoolVar:
 
 def _py_eq(a: Any, b: Any) -> bool:
     return a == b
+
+
+def not_eq(a: Any, b: Any) -> Union[bool, aq_types.BoolVar]:
+    """Functional form of "not equal".
+
+    Args:
+        a (Any): First expression to compare.
+        b (Any): Second expression to compare.
+
+    Returns:
+        Union[bool, BoolVar]: Whether the expressions are not equal.
+    """
+    if aq_types.is_qasm_type(a) or aq_types.is_qasm_type(b):
+        return _oqpy_not_eq(a, b)
+    else:
+        return _py_not_eq(a, b)
+
+
+def _oqpy_not_eq(a: Any, b: Any) -> aq_types.BoolVar:
+    oqpy_program = program.get_program_conversion_context().get_oqpy_program()
+    is_not_equal = aq_types.BoolVar()
+    oqpy_program.declare(is_not_equal)
+    oqpy_program.set(is_not_equal, a != b)
```

Comment:
The QASM types all support this? I'm surprised this one is not also `oqpy.base.something`

Reply:
Yes, `OQPyExpression` implements `__eq__` and `__ne__`: https://github.com/openqasm/oqpy/blob/main/oqpy/base.py#L156-L160